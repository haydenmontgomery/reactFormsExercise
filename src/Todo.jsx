import React, {useState} from "react";
import { v4 as uuidv4 } from 'uuid';

const Todo = ({idx, task, deleteTodo}) => {
  return (
    <div style={{
      position: "relative"
      }}>
      <li
      style={{
        position: "absolute",
        left: "80px",
        top: "10px"
      }}>{task}</li>
        <button 
          onClick={() => deleteTodo(idx)}
          style={{
            position: "absolute",
            textAlign: "center",
            left: "0px",
          }}>
          X
        </button>
    </div>
  )
}

export default Todo;