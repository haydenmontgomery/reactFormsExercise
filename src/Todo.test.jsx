import React from "react";
import { render, fireEvent } from "@testing-library/react";
import Todo from "./Todo";
import { it, expect } from "vitest";

it("renders without crashing", function () {
  render(<Todo />);
});

it("matches snapshot", function () {
  const { asFragment } = render(<Todo />);
  expect(asFragment()).toMatchSnapshot();
});