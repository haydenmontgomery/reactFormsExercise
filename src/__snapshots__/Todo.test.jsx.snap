// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    style="position: relative;"
  >
    <li
      style="position: absolute; left: 80px; top: 10px;"
    />
    <button
      style="position: absolute; text-align: center; left: 0px;"
    >
      X
    </button>
  </div>
</DocumentFragment>
`;
